tsdb/exemplar.go

```diff
@@ -19,12 +19,17 @@ import (
 	"sync"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/prometheus/pkg/exemplar"
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/storage"
 )
 
 type CircularExemplarStorage struct {
+	exemplarsAppended  prometheus.Counter
+	exemplarsInStorage prometheus.Gauge
+	//exemplarsLastTs        prometheus.Gauge
```

Comment:
Those are not easy to calculate, so not sure if worth adding, WDYT? @cstyan ?

Reply:
appended is easy enough, # of exemplars in storage I'm not sure about the usefulness, but last ts is hard to calculate. Do you mean the timestamp of the last exemplar we added, the timestamp we last added an exemplar at, the oldest timestamp of any exemplar stored, or the timestamp of the oldest exemplar? 

If you just wanted the timestamp for the exemplar that was stored the furthest in the past you could just export the timestamp of `buffer[nextIndex]`, but that doesn't necessarily mean it has the oldest timestamp out of all stored exemplars.

Reply:
Yes, Thanks for this. I added with proper help of metrics, let me know if that makes sense!